src/Services/AccessDefinitionAccessDeciderService.php

```diff
@@ -46,19 +46,21 @@ public function hasAccessDefinedAccess(?AccessDefinitionMetadata $metadata): boo
       return true;
     }
 
-    if ($this->shouldCheckScope()) {
-      return $metadata->scopes !== null && $this->security->isGranted(OauthAuthorizationVoter::OAUTH_SCOPE, $metadata->scopes);
+    if ($this->shouldCheckScope($metadata) && !$this->security->isGranted(OauthAuthorizationVoter::OAUTH_SCOPE, $metadata->scopes)) {
+      return false;
```

Comment:
This might be breaking.

Previous code: If we should check scope; ONLY check scope.
new code: check scopes and roles.

Not sure what we want here. Checking scopes and roles seams the most correct, but it might not work with client_credentials, since a client credentials user has no roles.

Reply:
As per conversation: We'll update the pack to support strategy (`affirmative` and `unanimous`) with `affirmative` being the default. This will ensure backwards compatibility, and enable us to achieve what we want here.

For this PR, we'll leave it as it is.